ethereum/smart-contract/contracts/GatewayNetwork.sol

```diff
@@ -0,0 +1,155 @@
+
+        bool isAlreadyGatekeeper = isGateKeeper(networkName, gatekeeper);
+
+        if(isAlreadyGatekeeper) {
+            revert GatewayNetworkGatekeeperAlreadyExists(string(abi.encodePacked(networkName)), gatekeeper);
+        }
+
+        GatekeeperNetworkData storage networkData = _networks[networkName];
+  
+        networkData.gatekeepers.push(gatekeeper);
+        emit GatekeeperNetworkGatekeeperAdded(gatekeeper);
+    }
+
+    function removeGatekeeper(address gatekeeper, bytes32 networkName) external override onlyPrimaryNetworkAuthority(networkName){
+        require(_networks[networkName].primaryAuthority != address(0), "Network does not exist");
+        bool isAlreadyGatekeeper = isGateKeeper(networkName, gatekeeper);
+
+        if(!isAlreadyGatekeeper) {
+            revert GatewayNetworkGatekeeperDoesNotExists(string(abi.encodePacked(networkName)), gatekeeper);
+        }
+
+        GatekeeperNetworkData storage networkData = _networks[networkName];
+        address[] storage currentGatekeepers = _networks[networkName].gatekeepers;
+
+        // Remove gatekeeper
+        for(uint i = 0; i < currentGatekeepers.length; i++) {
+            if(currentGatekeepers[i] == gatekeeper) {
+                // Swap gatekeeper to be removed with last element in the array
+                _networks[networkName].gatekeepers[i] = _networks[networkName].gatekeepers[currentGatekeepers.length - 1];
+                // Remove last element in array
+                _networks[networkName].gatekeepers.pop();
+            }
+        }
+
+        emit GatekeeperNetworkGatekeeperRemoved(gatekeeper);
+    }
+
+    // Two transactions are required to transfer the primary authority of a network
+    // Tx 1: The current primary authority submits a transaction setting the new primary authority
+    // Tx 2: The new primary authority must accept the update (this helps prevent misconfiguring the primary authority of a network)
+    function updatePrimaryAuthority(address newPrimaryAuthortiy, bytes32 networkName) external override onlyPrimaryNetworkAuthority(networkName) {
+        require(newPrimaryAuthortiy != address(0), "Primary authority cannot be set to the zero address");
+        _nextPrimaryAuthoritys[networkName] = newPrimaryAuthortiy;
+    } 
+
+    function claimPrimaryAuthority(bytes32 networkName) external override {
+        require(msg.sender == _nextPrimaryAuthoritys[networkName], "Can only claim authority on a network if given permission by the current primary authority");
+
+        _networks[networkName].primaryAuthority = msg.sender;
+
+        _nextPrimaryAuthoritys[networkName] = address(0);
+    } 
+
+    function updatePassExpirationTimestamp(uint newExpirationTimestamp, bytes32 networkName) external override onlyPrimaryNetworkAuthority(networkName) {
+        require(doesNetworkExist(uint(networkName)), "Network does not exist");
+        _networks[networkName].passExpireTimestamp = newExpirationTimestamp;
```

Comment:
If this is a global property of the network it needs to be a duration, rather then a timestamp. You can set a timestamp when minting a new Gateway Token, but the network itself cannot have a Timestamp expiry because it would be absolute in regard to any Gateway Token.

Reply:
Updated to configure a `passExpireTimeInSeconds`.